tests/baselines/reference/bestCommonTypeOfTuple2.types

```diff
@@ -29,7 +29,7 @@ class F extends C { f }
 class C1 implements base1 { i = "foo"; c }
 >C1 : C1
 >base1 : base1
->i : string
+>i : any

```

Comment:
This is a regression. `"foo"` should be contextually typed with `any`, but since that doesn't affect the type of `"foo"`, `i` should still infer its type as `string`.